File: ios/Plugin/VolumesPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(VolumesPlugin, "Volumes",
           CAP_PLUGIN_METHOD(getVolumeLevel, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(setVolumeLevel, CAPPluginReturnPromise);
)

File: ios/Plugin/VolumesPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(VolumesPlugin)
public class VolumesPlugin: CAPPlugin {
    private let implementation = Volumes()

    @objc func getVolumeLevel(_ call: CAPPluginCall) {
        call.resolve([
            "value": implementation.getVolumeLevel()
        ])
    }
    
    @objc func setVolumeLevel(_ call: CAPPluginCall) {
        let value = call.getFloat("value") ?? 5
        call.resolve([
            "value": implementation.setVolumeLevel(value)
        ])
    }
}
